### PR TITLE
[16.0][MIG] mail_drop_target: Migration to 16.0

### PR DESCRIPTION
**Pull request by nguyenminhchien**

### CODE REVIEW

File: mail_drop_target/controllers/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
```
Comment (grindtildeath) [inline mail_drop_target/controllers/__init__.py:1]: Please fix header

File: mail_drop_target/models/mail_thread.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            raise exceptions.UserError(
+                _("Install the msg-extractor library to handle .msg files")
+            )
+        message_msg = Message(b64decode(message))
+        try:
+            message_id = message_msg.messageId
+        except AttributeError:
+            # Using extract_msg < 0.24.4
+            message_id = message_msg.message_id
+        msg_body = message_msg.getStream("__substg1.0_10130102") or message_msg.body
```
Comment (grindtildeath) [inline mail_drop_target/models/mail_thread.py]: Would be great to have a comment explaining why this value in `getStream` looks so barbaric :sweat_smile:

**Review by grindtildeath — APPROVED** [review]
Thanks for the mig. Looks good, just a bit of nitpicking, approving anyway :wink:

**Review by houssine78 — APPROVED** [review]

File: mail_drop_target/models/mail_thread.py (hunk 2 above)
Comment (nguyenminhchien) [reply mail_drop_target/models/mail_thread.py]: Hi @grindtildeath , `message_msg.getStream("__substg1.0_10130102")` was removed.

**Comment by hbrunn** [comment]
@nguyenminhchien please fix conflicts

**Comment by nguyenminhchien** [comment]
> @nguyenminhchien please fix conflicts

fixed

**Review by hbrunn — APPROVED** [review]

**Comment by CRogos** [comment]
I am running Odoo in odoo.sh and since this module was added, I get a dependency version conflict which is originally caused by **extract_msg** . Does anyone have similar issues or an idea how so solve this?

```
2024-02-23T17:27:27,474 ERROR: Cannot install -r /home/odoo/.requirements.txt (line 14) and RTFDE because these package versions have conflicting dependencies.
2024-02-23T17:27:27,474 
2024-02-23T17:27:27,474 The conflict is caused by:
2024-02-23T17:27:27,474     pyrfc6266 1.0.0 depends on pyparsing~=3.0.7
2024-02-23T17:27:27,474     oletools 0.60.1 depends on pyparsing<3 and >=2.1.0
```

**Pull request merged**